drivers/can/can_stm32.c

```diff
@@ -409,6 +410,46 @@ static int can_stm32_init(const struct device *dev)
 		return -EIO;
 	}
 
+	/* configure pinmux */
+	if (cfg->pinctrl_len != 0U) {
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_pinctrl)
+		int remap;
+		/* Check that remap configuration is coherent across pins */
+		remap = stm32_dt_pinctrl_remap_check(cfg->pinctrl,
+						     cfg->pinctrl_len);
+		if (remap < 0) {
+			return remap;
+		}
+
+		/* A valid remapping configuration is provided */
+		/* Apply remapping before proceeding with pin configuration */
+		LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_AFIO);
+
+		switch ((uint32_t)cfg->can) {
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(can1), okay)
+		case DT_REG_ADDR(DT_NODELABEL(can1)):
+			if (remap == REMAP_FULL) { /* or REMAP_1 ??? */
+				LL_GPIO_AF_EnableRemap_CAN1();
+			} else {
+				LL_GPIO_AF_DisableRemap_CAN1();
```

Comment:
Looking at https://github.com/zephyrproject-rtos/hal_stm32/blob/master/stm32cube/stm32f1xx/drivers/include/stm32f1xx_ll_gpio.h, there seems to be

`LL_GPIO_AF_RemapPartial1_CAN1`, `LL_GPIO_AF_RemapPartial2_CAN1`, and `LL_GPIO_AF_RemapPartial3_CAN1`, even though they seem to only be available for certain families (LL API checks for `AFIO_MAPR_CAN_REMAP_REMAP1`).

Reply:
For some reason, there is no `LL_GPIO_AF_DisableRemap_CAN1`, even though all other remap-enabled peripherals like TIM and USART seem to have a corresponding `DisableRemap` function.
As the default register status is 0 (i.e. remap disabled), I guess keeping it at reset status is fine here?